types/ember-data/index.d.ts

```diff
@@ -271,7 +272,10 @@ export namespace DS {
      * subclasses are used to indicate specific error states. The following
      * subclasses are provided:
      */
-    class AdapterError extends Ember.Object {}
+    class AdapterError extends EmberError {
+        constructor();
```

Comment:
I don't think we should have the empty constructor form here. It may need to have a corresponding change removed from the test, or the test updated to include a message. As an `Error` subclass, it should accept a `message: string`. We get that for free that simply by not including this:

```suggestion
```

(If an existing test assumes that, it is likely the test which is wrong and needs updating.)